### ISSUE TITLE
When creating a project in ingest and not enough info, don't make the 'In Catalogue' checked by default

### ISSUE THREAD

**Issue by gabsie** [ebi-ait/dcp-ingest-central#570]
The problem: curators often in ingest backoffice populate just their names, a title and maybe not much more
(which results in project catalogue showing very incomplete projects)

So can we make the checkbox for Catalogue be unchecked by default, for newly created projects. 

If this can be made smarter, can we make it unchecked for projects missing: Title, Authors, organ, technology?

We can discuss more at our meeting.

**Comment by prabh-t** [comment]
@gabsie to talk @MightyAx through this ticket.

**Comment by MightyAx** [comment]
Probably what we want to be filled is:
- Title
- Authors

But I can also understand why we would want:
- Organs
- Technologies
- Cell Count

**Comment by MightyAx** [comment]
The above fields are already mandatory, but the "ignore validation" tick box allows them to be blank.
The current suggestion is filter out projects in the project-index if they have metadata-errors, which would mean the logic would be:

Projects are included in the catalogue if:
- the "include in catalogue" box is ticked
- **AND**
- there are no metadata errors

@ESapenaVentura, would that work?

**Comment by MightyAx** [comment]
Actually that won't work because half the catalogue entries have metadata errors :/
Now the plan is to clear the "include in catalogue" tickbox if the user ticks the "ignore validation errors" when creating new projects.

**Comment by ESapenaVentura** [comment]
@MightyAx I just had a chat with gabs, and I think it's mostly fine, but about point 2 `There are no metadata errors`, I would say that the projects that appear in the catalogue may have metadata errors, we just want to untick when there are metadata errors in the following fields:

- Title (`content.project_core.project_title`)
- Publication info (`content.publications.[]`, just check that there's at least 1 entry in the list)
- Organs (`organ`)
- Technology (`technology`)

Basically, we don't need all the required information in a project for it to appear in the catalogue. The DCP requires a lot more info than the project catalogue, and we just want to make sure that the "project catalogue-required" fields are in there.

**Comment by MightyAx** [comment]
We all chatted about the problems with the above implementations and have instead gone with the following:
This popup and functionality will only be triggered with new, invalid, projects that skip validation. 
![Screenshot 2021-12-20 at 12.45.13.png](https://images.zenhubusercontent.com/5d4053c054f6346037c2ef53/7fa4c85d-0c95-4914-b4f8-f0434eb19172)

**Comment by jacobwindsor** [comment]
This is in PR review

**Comment by ipediez** [comment]
There's a problem running the test in GitLab. Still to review

**Comment by MightyAx** [comment]
@amnonkhen to approve https://github.com/ebi-ait/ingest-ui/pull/146

**Comment by MightyAx** [comment]
Merged to Dev

**Comment by gabsie** [comment]
@gabsie can take a look at staging, as well as @ofanobilbao

**Comment by MightyAx** [comment]
@MightyAx to promote to staging after #575 is tested

**Comment by MightyAx** [comment]
@MightyAx to promote to staging

**Comment by MightyAx** [comment]
Testing finished on #575
@MightyAx to merge to master to trigger deployment to staging.

**Comment by MightyAx** [comment]
Deployed to **staging**, @ofanobilbao and @gabsie to test.

**Comment by MightyAx** [comment]
Tested on Staging:

https://user-images.githubusercontent.com/13031259/149301868-98c19940-adcb-4db4-943e-9da770c1b8c2.mov

**Comment by prabh-t** [comment]
Waiting for @ofanobilbao and @gabsie to test.

**Comment by Wkt8** [comment]
Oihane should be completing testing by today.

**Comment by gabsie** [comment]
Hi, everyone. Oihane and myself tested this and chatted about it.
What we discovered:

- Went and created a project, entered some minimal info, saved ignoring the validation, a message appears that it would not be in catalogue.
- Went and entered back some more info, including organ/technology and publication. Project should appear in catalogue, but it doesn't.
- We go back to check why and discover the checkbox for catalogue is unticked.
- We tick, the project appears. (With citation missing, but we know Jacob's script is going to put that in place overnight)

We feel the box should not be unticked on step one, but just ignored this time.. because it would be hard for a wrangler or anyone to remember to go and check for that as they enter more info. 
Additionally, not sure actually what parts of validation at the moment it is required to be passed for it to make it to catalogue (hopefully it's not ALL the validation boxes, but just organ/technology/Title/Publication)

Let us know if any questions.
Gabs

**Comment by ESapenaVentura** [comment]
All testing is done, @MightyAx to push this today

**Comment by ofanobilbao** [comment]
Adding to Gabby's last comment on this. I tested a scenario where I created a project from DOI, added the fields that we wanted for the catalogue (even more as I filled the entire first tab) but did not bother editing other tabs, like adding institutions for authors, or adding funders. In this scenario the system was giving me the validation error and removing it from the catalogue, even if all the fields that we wanted to display in the Catalogue were there

**Comment by gabsie** [comment]
@MightyAx will review and work on this today

**Comment by MightyAx** [comment]
> @gabs We feel the box should not be unticked on step one, but just ignored this time.. because it would be hard for a wrangler or anyone to remember to go and check for that as they enter more info.

Temporarily ignored is not a state that exists in the system, providing some way for wranglers to remember to go back and fill out more information for projects to be added into the catalogue feels like a different feature to this.

> @gabs Additionally, not sure actually what parts of validation at the moment it is required to be passed for it to make it to catalogue (hopefully it's not ALL the validation boxes, but just organ/technology/Title/Publication)

It is indeed all validation, that is the only method of validation we have. If the validation rules for the catalogue and ingest are different, more change will be required to create these validation rules, evaluate projects against these additional rules and store the results separately before we can act on them.

>  @ofanobilbao I created a project from DOI, added the fields that we wanted for the catalogue (even more as I filled the entire first tab) but did not bother editing other tabs, like adding institutions for authors, or adding funders. In this scenario the system was giving me the validation error and removing it from the catalogue, even if all the fields that we wanted to display in the Catalogue were there

As above there's is only one set of validation rules but in this scenario simply edit the project again to tick the "Add to Catalogue". My code change only applies when creating **new projects**. Editing existing projects you can add and remove projects from the catalogue unimpeded by their validation state.

These other aspects sound like great future enhancements, but aren't something we can easily provide within the scope of this ticket. I don't think these other aspects stop what I've built from being useful for the stated issue of preventing UCSC wranglers from adding empty projects to the catalogue, I hope you agree and we can promote this to production.
